src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,149 @@
+            var text = stringToken.ToString();
+
+            if (stringToken.IsVerbatimStringLiteral())
+            {
+                // We need to escape braces as this is an ambiguous case in interpolated strings that is 
+                // not ambiguous in verbatim strings
+                text = text.Replace(@"'{'", @"'{{'").Replace(@"'}'", @"'}}'");
+            }
+
+            return text;
+        }
+
+        protected override bool IsArgumentListCorrect(
+            InvocationExpressionSyntax invocation,
+            ISymbol invocationSymbol,
+            ImmutableArray<ISymbol> formatMethods,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            if (invocation.ArgumentList != null &&
+                invocation.ArgumentList.Arguments.Count >= 2 &&
+                invocation.ArgumentList.Arguments[0].Expression.IsKind(SyntaxKind.StringLiteralExpression))
+            {
+                // We do not want to substitute the expression if it is being passed to params array argument
+                // Example: 
+                // string[] args;
+                // String.Format("{0}{1}{2}", args);
+                return IsArgumentListNotPassingArrayToParams(
+                    invocation.ArgumentList.Arguments[1].Expression,
+                    invocationSymbol,
+                    formatMethods,
+                    semanticModel,
+                    cancellationToken);
+            }
+
+            return false;
+        }
+
+        protected override bool IsStringLiteral(LiteralExpressionSyntax firstArgument) =>
+            firstArgument?.Token.IsKind(SyntaxKind.StringLiteralToken) == true;
+
+        protected override InterpolatedStringExpressionSyntax VisitArguments(
+            ImmutableArray<ExpressionSyntax> expandedArguments,
+            InterpolatedStringExpressionSyntax interpolatedString)
+        {
+            return interpolatedString.ReplaceNodes(interpolatedString.Contents, (oldNode, newNode) =>
+            {
+                var node = newNode as InterpolationSyntax;
+                if (node == null)
+                {
+                    return newNode;
+                }
+
+                var literalExpression = node.Expression as LiteralExpressionSyntax;
+                if (literalExpression != null && literalExpression.IsKind(SyntaxKind.NumericLiteralExpression))
+                {
+                    var index = (int)literalExpression.Token.Value;
+                    if (index >= 0 && index < expandedArguments.Length)
+                    {
+                        return node.WithExpression(FixTrivia(expandedArguments[index]));
+                    }
+                }
+
+                return newNode;
+            });
+        }
+
+        /// <summary>
+        /// Since C# interpolations cannot be on more than one line, we need to remove newlines if possible.
```

Comment:
If the original expression was 
```C#
String.Format("{0}"
    args[0]);
```

I would argue that the developer would want the new expression to be 
`$"{args[0]}"` 

not
 
```
@$"{
    args[0]}"

---
In reply to: [65440670](https://github.com/dotnet/roslyn/pull/11415#discussion_r65440670) [](ancestors = 65440670)